compositor_common/src/scene/node.rs

```diff
@@ -30,10 +33,29 @@ pub enum NodeParams {
 }
 
 impl NodeSpec {
-    pub fn validate(&self) -> Result<(), NodeSpecValidationError> {
+    pub fn validate_params(&self) -> Result<(), NodeSpecValidationError> {
         match &self.params {
-            NodeParams::Builtin { transformation, .. } => Ok(transformation.validate(self)?),
+            NodeParams::Builtin { transformation } => Ok(transformation.validate_params(self)?),
             _ => Ok(()),
         }
     }
 }
+
+lazy_static! {
+    static ref TEXT_CONSTRAINTS: NodeConstraints = NodeConstraints(vec![Constraint::InputsCount(
+        InputsCountConstraint::Exact { fixed_count: 0 }
+    )]);
+    static ref IMAGE_CONSTRAINTS: NodeConstraints = NodeConstraints(vec![Constraint::InputsCount(
+        InputsCountConstraint::Exact { fixed_count: 0 }
+    )]);
+}
+
+impl NodeParams {
+    pub fn text_constraints() -> &'static NodeConstraints {
+        &TEXT_CONSTRAINTS
```

Comment:
I would just put it here instead of using lazy_static

Also make it return Vec instead